06-classification_continued.Rmd

```diff
@@ -559,16 +559,257 @@ The overall workflow for performing K-nearest neighbour classification using `ti
 
 **Strengths:**
 
-1. Simple and easy to understand
-2. No assumptions about what the data must look like 
-3. Works easily for binary (two-class) and multi-class (> 2 classes) classification problems
+1. Simple, intuitive algorithm
+2. Few assumptions about what the data must look like 
+3. Works for binary (two-class) and multi-class (> 2 classes) classification problems
 
 **Weaknesses:**
 
-1. As data gets bigger and bigger, K-nearest neighbour gets slower and slower, quite quickly
-2. Does not perform well with a large number of predictors
+1. As the training data gets larger, K-nearest neighbours becomes very slow
+2. May not perform well with a large number of predictors
 3. Does not perform well when classes are imbalanced (when many more observations are in one of the classes compared to the others)
 
+## Predictor variable selection
+
+> This section is not required reading for the remainder of the textbook.
+
+Another potentially important part of tuning your classifier is to choose which
+variables from your data will be treated as predictor variables. Based
+on your initial exploratory analysis, you can choose
+anything from using a single predictor variable to using every variable in your
+data; the K-nearest neighbours algorithm accepts any number of
```

Comment:
I am not sure what you mean here by "Based on your initial exploratory analysis", as you could (unwisely) put all the variables in your model without doing any exploratory analysis. I see exploratory analysis, in the realm of variable selection, as being useful to rule out any predictors that obviously have little to no predictive value. 

Exploratory analysis is also important for many other things of course (cataloguing missing values, examining class imbalance, feature engineering, etc), but maybe we can just leave it out of this sentence, and instead write:
```suggestion
variables from your data will be treated as predictor variables. 
Technically, you can choose
anything from using a single predictor variable to using every variable in your
data; the K-nearest neighbours algorithm accepts any number of
```

Reply:
agree